### DISCUSSION TITLE
registering a service - how to set the service executable?

### DISCUSSION THREAD

**Discussion by BryanCrotaz** [wixtoolset/issues#7812] [Questions]
I've managed to harvest my project files (on my dev machine, but not on the build server, that's another saga...) into a `ComponentGroupRef`. Now I want to register the exe as a service using `ServiceInstall`.

I *think* that you can only add `ServiceInstall` to a Component, so I need to nest my `ComponentGroupRef` inside a new `Component` and then attach the `ServiceInstall` tag to it. That seems easy enough -

```xml
<Component ...>
  <Fragment>
   <ComponentGroup Id="AgentComponents" Directory="INSTALLFOLDER"/>
  </Fragment>
  <ServiceInstall .../>
</Component>
```

How do you tell the `ServiceInstall` to reference the exe file? Or is that automatic? The docs talk about a `File` element but I don't know how to pull that out of the `ComponentGroupRef`. The docs say alternatively to use the `KeyPath` on the `Component` but the schema doesn't show a `KeyPath` property.

**Comment by barnson** [comment]
> The docs talk about a FileReference 

Where?

**Reply by BryanCrotaz** [reply]
Odd, I've edited my comment a while ago because I'd misremembered it. They mention a `File` element not `FileReference`.
https://wixtoolset.org/docs/schema/wxs/serviceinstall/

**Comment by BryanCrotaz** [comment]
Another thought, can I have two harvests, one that gets the exe and one that gets everything else? I can't see how to put any exclusions on a `HarvestDirectory`, e.g. to only harvest dlls
